src/lib/optimizer/strategy/chunk_pruning_rule.cpp

```diff
@@ -90,21 +90,60 @@ std::set<ChunkID> ChunkPruningRule::_compute_exclude_list(
     if (!is_variant(operator_predicate.value)) {
       return std::set<ChunkID>();
     }
+
     const auto& value = boost::get<AllTypeVariant>(operator_predicate.value);
+
     std::optional<AllTypeVariant> value2;
-    if (static_cast<bool>(operator_predicate.value2)) value2 = boost::get<AllTypeVariant>(*operator_predicate.value2);
+    if (static_cast<bool>(operator_predicate.value2)) {
+      value2 = boost::get<AllTypeVariant>(*operator_predicate.value2);
+    }
+
     auto condition = operator_predicate.predicate_condition;
-    for (auto chunk_id = ChunkID{0}; chunk_id < statistics.size(); ++chunk_id) {
-      // statistics[chunk_id] can be a shared_ptr initialized with a nullptr
-      if (statistics[chunk_id] &&
-          statistics[chunk_id]->can_prune(operator_predicate.column_id, condition, value, value2)) {
-        result.insert(chunk_id);
+
+    for (auto chunk_id = ChunkID{0}; chunk_id < table.chunk_count(); ++chunk_id) {
+      const auto pruning_statistics = table.get_chunk(chunk_id)->pruning_statistics();
+      if (!pruning_statistics) {
+        continue;
+      }
+
+      const auto segment_statistics = (*pruning_statistics)[operator_predicate.column_id];
+      if (_can_prune(*segment_statistics, condition, value, value2)) {
+        result.insert(ChunkID(chunk_id));
       }
     }
   }
   return result;
 }
 
+bool ChunkPruningRule::_can_prune(const BaseColumnStatistics& base_column_statistics,
+                                  const PredicateCondition predicate_condition, const AllTypeVariant& variant_value,
+                                  const std::optional<AllTypeVariant>& variant_value2) const {
+  auto any_filter_prunes = false;
```

Comment:
This is just the return value in a way where it can be written by the lambda, correct? If so, maybe make this clear by calling it `can_prune`, same as the method?